lib/Middleware/ErrorHandlingMiddleware.php

```diff
@@ -55,13 +49,14 @@ public function process(Message $request, RequestHandler $handler): Promise
                 ));
             } catch (Throwable $error) {
                 $message = sprintf('Exception [%s] %s', get_class($error), $error->getMessage());
+                $this->logger->error(sprintf(
+                    'Error when handling "%s" (%s): %s',
+                    get_class($request),
+                    json_encode($request),
+                    $message
+                ));
+
                 if (!$request instanceof RequestMessage) {
```

Comment:
hm, the reasoning here was probably that the client receives the error message, so it's not thrown to the wind.

but having consistent logging sounds good.